### PR TITLE
Add PHP 7.3.5

### PR DESCRIPTION
**Pull request by pyhalov**

### CODE REVIEW

**Comment by pyhalov** [comment]
Notes and changes from 7.0: 
- PHP mcrypt module was removed;
- we need external libzip to enable additional zip module features;
- basic extensions are merged to php-common package.

**Comment by Toasterson** [comment]
I would love if we could switch php-fpm to 64bit by default. I am already hitting limits with the 32bit version and nextcloud. 32bit could stay as a alternative or be removed but 64bit php should be the default

**Comment by pyhalov** [comment]
I think we can deliver only 64-bit-only PHP (and this component actually does this).

File: components/web/php/php-7_3/Makefile [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+#
+
+#
+# Copyright 2013, EveryCity Ltd. All rights reserved.
+#
+include ../../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		php
+COMPONENT_MJR_VERSION=	7.3
+COMPONENT_MNR_VERSION=	4
```
Comment (Mno-hime) [inline components/web/php/php-7_3/Makefile]: There's minor update to 7.3.5: https://www.php.net/ChangeLog-7.php#7.3.5.

File: components/web/php/php-7_3/Makefile [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+CXXFLAGS	+=	$(JPEG_CPPFLAGS) $(JPEG_CXXFLAGS)
+CPPFLAGS	+=	$(JPEG_CPPFLAGS)
+LDFLAGS		+=	$(JPEG_LDFLAGS)
+
+COMPONENT_PREP_ACTION =        ( cd $(@D) && autoconf )
+
+# Configure Options
+CONFIGURE_OPTIONS  =	--prefix=$(PHP_PREFIX)
+CONFIGURE_OPTIONS  +=	--bindir=$(PHP_PREFIX)/bin
+CONFIGURE_OPTIONS  +=	--sbindir=$(PHP_PREFIX)/bin
```
Comment (Mno-hime) [inline components/web/php/php-7_3/Makefile:58]: `sbin`?

File: components/web/php/php-7_3/Makefile [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+#CONFIGURE_OPTIONS     +=	--with-libexpat-dir=$(ECPREFIX)
+#CONFIGURE_OPTIONS     +=	--with-openssl-dir=$(ECPREFIX)
+
+# Disabled extensions & Options
+CONFIGURE_OPTIONS   +=	--disable-static
+CONFIGURE_OPTIONS   +=	--disable-dba
+CONFIGURE_OPTIONS   +=	--disable-debug
+CONFIGURE_OPTIONS   +=	--disable-libgcc
+
+CONFIGURE_OPTIONS   +=	--enable-dtrace
```
Comment (Mno-hime) [inline components/web/php/php-7_3/Makefile]: Move it to "Enabled extensions" section?

File: components/web/php/php-7_3/files/php_config.ini [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#if defined(_LP64) || defined (__amd64)
+#include <main/php_config-64.h>
+#else
+#include <main/php_config-32.h>
```
Comment (Mno-hime) [inline components/web/php/php-7_3/files/php_config.ini]: Do we still ship any of these two? I see only `main/php_config.h`.

File: components/web/php/php-7_3/Makefile [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+CONFIGURE_ENV +=     EXTENSION_DIR=$(PHP_PREFIX)/extensions
+CONFIGURE_ENV.64 +=     ICU_CONFIG=/usr/bin/amd64/icu-config
+CONFIGURE_ENV +=	CPPFLAGS="$(CPPFLAGS)"
+CONFIGURE_ENV +=	PG_CONFIG=$(PG_BINDIR)/pg_config
+
+build:		$(BUILD_64)
+
+install:	$(INSTALL_64)
+
+# Build dependencies
```
Comment (Mno-hime) [inline components/web/php/php-7_3/Makefile:212]: What about `test:` section?

File: components/web/php/php-7_3/files/php_config.ini (hunk 4 above)
Comment (pyhalov) [reply components/web/php/php-7_3/files/php_config.ini]: No.

File: components/web/php/php-7_3/Makefile (hunk 2 above)
Comment (pyhalov) [reply components/web/php/php-7_3/Makefile:58]: no, we don't have /usr/php/X.Y/sbin, just bin.

**Review by Mno-hime — APPROVED** [review]

File: components/web/php/php-7_3/Makefile [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+	  echo "\# LoadModule foo_module libexec/mod_foo.so" >> $(PROTO_DIR)/$(AP_SYSCONFDIR)/conf.d/modules.load ; \
+	  echo "\# " >> $(PROTO_DIR)/$(AP_SYSCONFDIR)/conf.d/modules.load ; \
+	  echo "LoadModule foo_module libexec/mod_foo.so " >> $(PROTO_DIR)/$(AP_SYSCONFDIR)/conf.d/modules.load ; \
+          cp $(PROTO_DIR)/$(AP_SYSCONFDIR)/conf.d/modules.load  $(PROTO_DIR)/$(AP_SYSCONFDIR)/httpd.conf \
+)
+
+# Environment Variables
+COMPONENT_BUILD_ENV+=	CC="$(CC) $(CC_BITS)" CXX="$(CXX)" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)"
+CONFIGURE_ENV +=     EXTENSION_DIR=$(PHP_PREFIX)/extensions
+CONFIGURE_ENV.64 +=     ICU_CONFIG=/usr/bin/amd64/icu-config
```
Comment (Toasterson) [inline components/web/php/php-7_3/Makefile:185]: Should we switch that to CONFIGURE_ENV only?

**Review by Toasterson — APPROVED** [review]

File: components/web/php/php-7_3/Makefile (hunk 6 above)
Comment (pyhalov) [reply components/web/php/php-7_3/Makefile:185]: well, it's too amd64-specific, don't know.

Comment (Toasterson) [reply components/web/php/php-7_3/Makefile:185]: True. Ship it

**Pull request merged**